File: Data Structures & Algorithms/duplicate-integer/submission-3.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
      if (nums.empty()){
        return false;
      }

      std::sort(nums.begin(), nums.end());
      for (auto i=0;i<nums.size() -1;i++){
        if (nums[i] == nums[i+1]){
            return true;
        }
      }
      return false;  
    }
};

File: Data Structures & Algorithms/reconstruct-flight-path/submission-3.py
class Solution:
    def findItinerary(self, tickets: List[List[str]]) -> List[str]:
        adj = defaultdict(list)
        tickets.sort( reverse=True)
        for u,v in tickets:
            adj[u].append(v)
        res = []
        def dfs(u):
            while adj[u]:
                dfs(adj[u].pop())                
            res.append(u)
        dfs('JFK')
        return res[::-1]
        